src/sediment_model.jl

```diff
@@ -153,6 +158,41 @@ function initialize_sediment_model(config::Config)
     return model
 end
 
+function init_overland_flow_sediment(river::Vector, number_of_cells::Integer)
+    river_cell = convert(Vector{Float}, river)
+
+    overland_flow_sediment = OverlandFlowSediment{Float}(;
```

Comment:
Instead of `Float` you could also use `T` here, equal to the `SBM` model concept.

Reply:
I added an argument to the function call to generalize the construction of `OverlandModelSediment`.
It can now construct OverlandModelSediment with type parameters other than `Float` as long as it is a subtype of `AbstractFloat`.